### PR TITLE
docs(jenkins): Add documentation for unit testing prescriptive guidance

### PR DESCRIPTION
**Pull request by soda480**
Add documentation describing prescriptive guidance for facilitating unit testing of Jenkins declarative pipeline logic.

Signed-off-by: Emilio Reyes <emilio.reyes@intel.com>

## PR Checklist
Please check if your PR fulfills the following requirements:

- [ ] The commit message follows our guidelines: https://github.com/edgexfoundry/ci-management/blob/master/.github/CONTRIBUTING.md
- [ ] Tests for the changes have been added (for bug fixes / features)
- [X] Added labels
## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [X] Documentation content changes
- [ ] Other... Please describe:

## Issue Number:
Fixes https://github.com/edgexfoundry/edgex-global-pipelines/issues/194

## Sandbox Testing
NA

### CODE REVIEW

**Review by cjoyv — DISMISSED** [review]
LGTM

**Review by bill-mahoney — APPROVED** [review]
LGTM

File: docs_src/tutorials/unit-testing-guidance.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Unit Testing Prescriptive Guidance
+
+The edgex-global-pipelines shared library leverages the [Jenkins Spock framework](https://github.com/ExpediaGroup/jenkins-spock) for unit testing Jenkins pipeline scripts and functions. The Jenkins Spock unit test framework does not currently support unit testing of Jenkins declarative pipeline code.
```
Comment (ernestojeda) [inline docs_src/tutorials/unit-testing-guidance.md:3]: Can you add a link to declarative pipeline when you mention "[Jenkins declarative pipeline](https://www.jenkins.io/doc/book/pipeline/syntax/#declarative-pipeline)"

File: docs_src/tutorials/unit-testing-guidance.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+# Unit Testing Prescriptive Guidance
+
+The edgex-global-pipelines shared library leverages the [Jenkins Spock framework](https://github.com/ExpediaGroup/jenkins-spock) for unit testing Jenkins pipeline scripts and functions. The Jenkins Spock unit test framework does not currently support unit testing of Jenkins declarative pipeline code. 
+
+## Encapsulate Pipeline logic witin Groovy functions
+In order to facilitate unit testing of the edgex-global-pipelines shared library, the DevOps team has made a deliberate effort to to minimize the amount of scripting logic contained within Jenkins declarative pipelines. This is accomplished by encapsulating pipeline logic within a Groovy function and calling the function in the declarative pipeline step as needed. Localizing pipeline logic within Groovy functions enables the Jenkins Spock framework to provide greater test coverage of Pipeline logic.
```
Comment (ernestojeda) [inline docs_src/tutorials/unit-testing-guidance.md:6]: Can we get an example of this?

**Review by ernestojeda — CHANGES_REQUESTED** [review]
An example would be nice here

File: docs_src/tutorials/unit-testing-guidance.md (hunk 2 above)
Comment (soda480) [reply docs_src/tutorials/unit-testing-guidance.md:6]: done

**Review by bill-mahoney — DISMISSED** [review]
LGTM, I like the example too

**Review by ernestojeda — APPROVED** [review]
Looks great. Thanks for the example.

**Pull request merged**